### PR TITLE
[BugFix] do not support create list partition cloud table

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
report an error if user tries to create list partition cloud table

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/lake/CreateLakeTableTest.java [hunk 1]
```diff
@@ -295,6 +295,14 @@ public void testCreateLakeTableException() {
...
                                 "distributed by hash(key1) buckets 3\n" +
                                 " properties('datacache.enable' = 'false', 'enable_async_write_back' = 'true');"));
+
+        // do not support list partition
+        ExceptionChecker.expectThrowsWithMsg(DdlException.class,
+                "Do not support create list partition Cloud Native table",
+                () -> createTable(
+                        "create table lake_test.list_partition_invalid (dt date not null, key2 varchar(10))\n" +
+                                "distributed by hash(dt) buckets 3\n" +
+                                "PARTITION BY LIST (dt) (PARTITION p1 VALUES IN (("2022-04-01")),PARTITION p2 VALUES IN (("2022-04-02")),PARTITION p3 VALUES IN (("2022-04-03")));"));
```
Comment (wyb) [inline fe/fe-core/src/test/java/com/starrocks/lake/CreateLakeTableTest.java:305]: partition by first, then distributed by

**Review by kevincai — APPROVED** [review]

File: fe/fe-core/src/test/java/com/starrocks/lake/CreateLakeTableTest.java (hunk 1 above)
Comment (starrocks-xupeng) [reply fe/fe-core/src/test/java/com/starrocks/lake/CreateLakeTableTest.java]: done

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 10 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|6|9|66.67%|[346, 348, 349]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 10 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|6|9|66.67%|[346, 348, 349]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 9 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|6|9|66.67%|[346, 348, 349]|

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**